### DISCUSSION TITLE
error: expected ‘)’ before ‘(’ token when compiling on centos/rocky 8

### DISCUSSION THREAD

**Discussion by OrenMustaki** [pybind/pybind11#3642] [Q&A]
Hi 
Im getting a compilation error when im trying to compile on CentOS 8.4 ( same on rocky 8.5 )
It works fine when compiling on CentOS 7

a simple example to reproduce the error :
test.cpp:

      #include <pybind11/embed.h>
      #include "test.hpp"
      namespace test {
      
        void bind_test(pybind11::module& m){
            using namespace::pybind11::literals;
        
            pybind11::class_<BindingClass>(m, "Test")
            .def_readonly_static("SNAN", &SNAN)
        }
      }
      PYBIND11_EMBEDDED_MODULE(_test, m){
        m.doc() = "test";
        test::bind_test(m);
      }

test.hpp

        #ifndef TEST_HPP
        #define TEST_HPP
        
        #include <limits>
        namespace test{
            class BindingClass{};
        
            static constexpr double SNAN = std::numeric_limits<double>::signaling_NaN();
        }
        #endif

to compilr run:

g++ test.cpp -c -isystem /path/to/python/3.9.7/include/python3.9/ -isystem /path/to/pybind11/2.9.0/include/

the compilation fails with intel compiler ( OneApi 2022.1.1 ) and with python 3.7.2

any ideas ?

thanks in advance

Oren

**Comment by wallds** [comment]
![image](https://user-images.githubusercontent.com/12907032/151159486-08962996-dcf8-4061-b106-dd9a28014ab0.png)
It seems to be that a semicolon is missing here.

**Reply by OrenMustaki** [reply]
ive just tried it but im still getting the same error
